service/dap/server.go

```diff
@@ -106,6 +106,28 @@ var DefaultLoadConfig = proc.LoadConfig{
 	MaxStructFields:    -1,
 }
 
+// tempBinary offers a thread-safe abstraction for
+// managing a temp compiled binary (if any).
+type tempBinary struct {
+	mux      sync.Mutex
```

Comment:
Is there a reason to have a separate mutex for binary? 
Given that the lifetime of the binary is bound to the debugger's, I think it makes more sense to share a single lock. (Server's mu, for example). I expect to see guard against other fields  in the feature (isn't s.debugger or others need protection too?). Fine-grained locks are hard to use and can lead to deadlock if misused, it's better to avoid them if they are really necessary.

Reply:
You are right. I could reuse mu. I will make the change.